sklearn/utils/validation.py

```diff
@@ -94,6 +93,19 @@ def inner_f(*args, **kwargs):
     return _inner_deprecate_positional_args
 
 
+def _object_dtype_any_isnan(X):
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", DeprecationWarning)
+        try:
+            return (X != X).any()
+        except (DeprecationWarning, TypeError):
+            # Deprecation warning happens because pd.NA uses three-valued logic
```

Comment:
For other reviewers, I'm not completely sold on this implementation, but I was not able to find a better way to keep the original runtime speed and also catch `pd.NA`.